config.c

```diff
@@ -34,6 +34,13 @@
 
 #define REALLOC_STEP    10
 
+/* The __unused__ attribute was added in gcc 3.2.7.  */
+#if __GNUC__ >= 3 || (__GNUC__ == 2 && __GNUC_MINOR__ >= 7)
+# define ATTRIBUTE_UNUSED __attribute__((__unused__))
+#else
+# define ATTRIBUTE_UNUSED /* empty */
+#endif
+
```

Comment:
What exactly is the problem with the following one-line fix I proposed?

```
--- a/config.c
+++ b/config.c
@@ -710,6 +710,8 @@ int readAllConfigPaths(const char **paths)

 static int globerr(const char *pathname, int theerr)
 {
+    (void) pathname;
+
     /* A missing directory is not an error, so return 0 */
     if (theerr == ENOTDIR)
         return 0;
```

The fix you are proposing is way more complicated and would work with GCC only anyway.